osu.Game/Screens/Select/BeatmapPlaylistItem.cs

```diff
@@ -0,0 +1,311 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Bindables;
+using osu.Framework.Extensions.Color4Extensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Colour;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Input.Events;
+using osu.Framework.Localisation;
+using osu.Game.Beatmaps.Drawables;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Containers;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Online.Multiplayer;
+using osuTK;
+using osuTK.Graphics;
+using osuTK.Input;
+
+namespace osu.Game.Screens.Select
+{
+    public class BeatmapPlaylistItem : Container
+    {
+        public readonly Bindable<PlaylistItem> PlaylistItem = new Bindable<PlaylistItem>();
+
+        public event Action<BeatmapPlaylistItem> RequestRemoval;
+
+        private const int fade_duration = 60;
+        private readonly DragHandle dragHandle;
+        private readonly RemoveButton removeButton;
+        private bool isHovered;
+        private bool isDragged;
+
+        public BeatmapPlaylistItem(PlaylistItem item)
+        {
+            UpdateableBeatmapBackgroundSprite cover;
+            Height = 50;
+            RelativeSizeAxes = Axes.X;
+            Children = new Drawable[]
```

Comment:
this children block is verging on the point where i'd look to extract specific pieces out into their own components. even if those components are just nested classes.